<?xml version="1.0"?>
<launch>
    <node pkg="tf" type="static_transform_publisher" name="iris_0_base_link_to_laser"
    args="0.0 0.0 0.06 0.0 0.0 0.0 /base_link /iris_0/laser_2d 40" />
    <!-- MAVROS posix SITL environment launch script -->
    <!-- launches MAVROS, PX4 SITL, Gazebo environment, and spawns vehicle -->
    <!-- vehicle pose -->
    <arg name="x" default="0"/>
    <arg name="y" default="0"/>
    <arg name="z" default="0.5"/>
    <arg name="R" default="0"/>
    <arg name="P" default="0"/>
    <arg name="Y" default="1"/>
    <!-- vehicle model and world -->
    <arg name="est" default="ekf2"/>
    <arg name="vehicle" default="iris"/>
    <arg name="world" default="$(find multi_rotor_avoidance_rl)/worlds/env2.world"/>
    <arg name="sdf" default="$(find mavlink_sitl_gazebo)/models/falco_01/urdf/falco_01.sdf"/>

    <!-- gazebo configs -->
    <arg name="gui" default="true"/>
    <arg name="debug" default="false"/>
    <arg name="verbose" default="false"/>
    <arg name="paused" default="false"/>
    <arg name="respawn_gazebo" default="false"/>
    <!-- MAVROS configs -->
    <arg name="fcu_url" default="udp://:24540@localhost:34580"/>
    <arg name="respawn_mavros" default="false"/>
    <!-- PX4 configs -->
    <arg name="interactive" default="true"/>
    <!-- PX4 SITL and Gazebo -->
    <include file="$(find px4)/launch/posix_sitl.launch">
        <arg name="x" value="$(arg x)"/>
        <arg name="y" value="$(arg y)"/>
        <arg name="z" value="$(arg z)"/>
        <arg name="R" value="$(arg R)"/>
        <arg name="P" value="$(arg P)"/>
        <arg name="Y" value="$(arg Y)"/>
        <arg name="world" value="$(arg world)"/>
        <arg name="vehicle" value="$(arg vehicle)"/>
        <arg name="sdf" value="$(arg sdf)"/>
        <arg name="gui" value="$(arg gui)"/>
        <arg name="interactive" value="$(arg interactive)"/>
        <arg name="debug" value="$(arg debug)"/>
        <arg name="verbose" value="$(arg verbose)"/>
        <arg name="paused" value="$(arg paused)"/>
        <arg name="respawn_gazebo" value="$(arg respawn_gazebo)"/>
    </include>
    <!-- MAVROS -->
    <group ns="iris_0">
         <node pkg="mavros" type="mavros_node" name="mavros" output="screen" >
            <param name="fcu_url" value="$(arg fcu_url)" />
            <param name="gcs_url" value="" />
            <param name="target_system_id" value="1" />
            <param name="target_component_id" value="1" />
            <rosparam command="load" file="$(find multi_rotor_avoidance_rl)/launch/px4_pluginlists.yaml" />
            <rosparam command="load" file="$(find multi_rotor_avoidance_rl)/launch/px4_config.yaml" />
        </node>
    </group>
   
    <node pkg="laser_filters" type="scan_to_scan_filter_chain" name="laser_filter">
        <rosparam command="load" file="$(find multi_rotor_avoidance_rl)/launch/range_filter_sim.yaml" />
        <remap from="scan" to="/iris_0/scan" />
        <remap from="scan_filtered" to="/iris_0/scan_filtered" />
    </node>


    <node pkg="multi_rotor_avoidance_rl" type="laser_filter_downsample.py" name="laser_filter_downsameple" />

</launch>
